src/rgw/rgw_xxh_digest.h

```diff
@@ -0,0 +1,53 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2023 Red Hat, Inc
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stdio.h>
+#include "rgw_crc_digest.h"
+
+#define XXH_INLINE_ALL 1 /* required for streaming variants */
+#include "xxhash.h"
```

Comment:
i'm seeing compilation failures here and that define looks related? with gcc 13.2.1
```
In file included from /home/cbodley/ceph/src/rgw/rgw_xxh_digest.h:23,
                 from /home/cbodley/ceph/src/rgw/rgw_cksum.h:30,
                 from /home/cbodley/ceph/src/rgw/rgw_cksum_pipe.h:21,
		 from /home/cbodley/ceph/src/rgw/rgw_cksum_pipe.cc:16:
In function ‘void XXH3_accumulate(xxh_u64*, const xxh_u8*, const xxh_u8*, size_t, XXH3_f_accumulate_512)’,
    inlined from ‘void XXH3_hashLong_internal_loop(xxh_u64*, const xxh_u8*, size_t, const xxh_u8*, size_t, XXH3_f_accumulate_512, XXH3_f_scrambleAcc)’ at /home/cbodley/ceph/src/xxHash/xxhash.h:4391:24,
    inlined from ‘XXH64_hash_t XXH3_hashLong_64b_internal(const void*, size_t, const void*, size_t, XXH3_f_accumulate_512, XXH3_f_scrambleAcc)’ at /home/cbodley/ceph/src/xxHash/xxhash.h:4454:32,
    inlined from ‘XXH64_hash_t XXH3_hashLong_64b_withSecret(const void*, size_t, XXH64_hash_t, const xxh_u8*, size_t)’ at /home/cbodley/ceph/src/xxHash/xxhash.h:4474:38,
    inlined from ‘XXH64_hash_t XXH3_64bits_internal(const void*, size_t, XXH64_hash_t, const void*, size_t, XXH3_hashLong64_f)’ at /home/cbodley/ceph/src/xxHash/xxhash.h:4555:22,
    inlined from ‘XXH64_hash_t XXH_INLINE_XXH3_64bits_withSecret(const void*, size_t, const void*, size_t)’ at /home/cbodley/ceph/src/xxHash/xxhash.h:4571:32:
/home/cbodley/ceph/src/xxHash/xxhash.h:3932:1: error: inlining failed in call to ‘always_inline’ ‘void XXH3_accumulate_512_sse2(void*, const void*, const void*)’: function not considered for inlining
 3932 | XXH3_accumulate_512_sse2( void* XXH_RESTRICT acc,
      | ^~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
rrrr

Reply:
you need the define to expose the required defns;  I'll work on it

Reply:
this seems to be a known bug [1] ; gcc-13.2 compile completed successfully after I pulled xxHash up to .0.8.2
[1] https://github.com/Cyan4973/xxHash/issues/800

Reply:
works for me now 👍

Reply:
i saw that epel9 has xxhash 0.8.2 but ubuntu jammy only has 0.8.1. so close to being able to drop the submodule 🤷